### ISSUE TITLE
Issue with ConfigResourceClientAutoConfiguration Not autowiring correct RestTemplate

### ISSUE THREAD

**Issue by mcordeiro73** [pivotal-cf/spring-cloud-services-starters#412]
### Describe the bug

After upgrading to version 4.1.1m receive an error at application startup in any application that defines a RestTemplate Bean.  This appears to be caused by ConfigResourceClientAutoConfiguration being unable to identify the correct RestTemplate to be Autowired in.

Here is an example error being received:
```
***************************
APPLICATION FAILED TO START
***************************

Description:

Parameter 0 of method configResourceClient in io.pivotal.spring.cloud.config.client.ConfigResourceClientAutoConfiguration required a single bean, but 2 were found:
	- FASTRESTTemplate: defined by method 'fastRestTemplate' in class path resource [com/example/service/AppConfiguration.class]
	- configClientRestTemplate: a programmatically registered singleton
```

### Reproduction steps

1.  Create app with a custom RestTemplate Bean defined via a Bean method in a Configuration class
2. Attempt to start application in Tanzu with a spring-cloud-config-service service bound to that app.

### Expected behavior

ConfigResourceClientAutoConfiguration is able to properly identify and Autowire in the correct RestTemplate.

**Comment by stil4m** [comment]
I'm running in the same issue.

**Comment by kvmw** [comment]
@mcordeiro73 @stil4m , Thank you for reporting the issue. 
I've merged a PR which fixes the issue. Would you be able to build the latest code, locally, and verify the fix?

**Comment by stil4m** [comment]
@kvmw 
Just installed locally, build and pushed our product to Cloud Foundry again and it works 👍 .

**Comment by kvmw** [comment]
@stil4m  Awesome, Thank you.  
I'll cut a release today.

**Comment by kvmw** [comment]
Fixed in 4.1.2
